services/skus/service.go

```diff
@@ -2282,6 +2292,122 @@ func (s *Service) processSubmitReceipt(ctx context.Context, req model.ReceiptReq
 	return rcpt, nil
 }
 
+const errUnknownAction = model.Error("skus: unknown action") // TODO better message and rename
+
+func (s *Service) processRadomEvent(ctx context.Context, event radom.Event) error {
+	if !event.ShouldProcess() {
+		return nil
+	}
+
+	tx, err := s.Datastore.RawDB().BeginTxx(ctx, nil) // TODO
+	if err != nil {
+		return err
+	}
+	defer func() { _ = tx.Rollback() }()
+
+	if err := s.processRadomEventTx(ctx, tx, event); err != nil {
+		return err
+	}
+
+	return tx.Commit()
+}
+
+func (s *Service) processRadomEventTx(ctx context.Context, dbi sqlx.ExtContext, event radom.Event) error {
+	switch {
+	case event.IsNewSub():
+		oid, err := event.OrderID()
+		if err != nil {
+			return err
+		}
+
+		subID, err := event.SubID()
+		if err != nil {
+			return err
+		}
+
+		rsub, err := s.radomClient.GetSubscription(ctx, subID)
+		if err != nil {
+			return err
+		}
+
+		nxtB, err := rsub.NextBillingDate()
+		if err != nil {
+			return err
+		}
+
+		expAt := nxtB.Add(24 * time.Hour)
```

Comment:
How is this expAt value being stored and used? Essentially what I'm trying to narrow down here is if this expiration time leaks through to the blinded session tokens such that a token issued to a user for access to a service could be linked to a payment processor timing. If it is we'd need to look more deeply into how to prevent this temporal correlation by changing the time we add here, but first I'd just like to better understand how it's passed around and used within our services.

I see this is a a pattern that's been used for other service providers as well. Was this previously reviewed by chance?

Reply:
The `expAt` value adds 24 hours onto the expiry time in case our system is down and we cant process the webhooks essentially we are adding a grace period. IIRC tokens also have a grace so added so I am not sure a correlation could be drawn.

Reply:
I took a look at this further after a discussion with @evq to make sure I understood things correctly. For TLV2 creds this definitely isn't an issue since the solution is designed around expiration occurring by no longer being issued credentials all together. As for TLV1 credentials, this also isn't an issue because of the way we [timeChunk via an interval](https://github.com/brave-intl/bat-go/blob/fc0db24c865959edc12f0c3bd3937edbb85eb19a/services/skus/service.go#L1268-L1299) which creates a distribution that should be well enough distributed for us given we're using 1 month intervals in practice for the systems still relying on TLV1.